### PR TITLE
okhttp: Enable TLS for Http2OkHttpTest.

### PR DESCRIPTION
**Pull request by madongfly**
@ejona86

### CODE REVIEW

File: integration-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java [hunk 1]
```diff
@@ -236,41 +225,11 @@ protected ChannelImpl createChannel() {
...
-          // Force the hostname to match the cert the server uses.
-          builder.overrideHostForAuthority(serverHostOverride);
-        }
         if (useTls) {
-          try {
-            builder.sslSocketFactory(getSslSocketFactory());
-          } catch (Exception e) {
-            throw new RuntimeException(e);
-          }
+          Util.configureTlsForOkHttpChannel(builder, serverHostOverride, "ca.pem");
```
Comment (ejona86) [inline integration-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java]: You're no longer checking useTestCa.

Comment (madongfly) [inline integration-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java]: Good catch!

Done.

**Comment by madongfly** [comment]
@ejona86 PTAL, thanks!

File: integration-testing/src/main/java/io/grpc/testing/integration/Util.java [hunk 2]
```diff
@@ -119,4 +131,42 @@ public static void assertEquals(List<? extends MessageLite> expected,
       }
     }
   }
+
+  /**
+   * Configure an OkHttpClientChannelBuilder so that it can use TLS correctly.
+   */
+  public static void configureTlsForOkHttpChannel(OkHttpChannelBuilder channelBuilder,
```
Comment (ejona86) [inline integration-testing/src/main/java/io/grpc/testing/integration/Util.java]: This method seems to do a couple of different things. Could it simply accept a non-null certFile and return the SSLSocketFactory?

Comment (madongfly) [inline integration-testing/src/main/java/io/grpc/testing/integration/Util.java]: Done.

File: integration-testing/src/test/java/io/grpc/testing/integration/Http2OkHttpTest.java [hunk 3]
```diff
@@ -59,6 +69,13 @@ public static void stopServer() throws Exception {
 
   @Override
   protected ChannelImpl createChannel() {
-    return OkHttpChannelBuilder.forAddress("127.0.0.1", serverPort).build();
+    OkHttpChannelBuilder builder = OkHttpChannelBuilder.forAddress("127.0.0.1", serverPort);
+    builder.overrideHostForAuthority("foo.test.google.fr");
```
Comment (ejona86) [inline integration-testing/src/test/java/io/grpc/testing/integration/Http2OkHttpTest.java]: Chain this?

```java
OkHttpChannelBuilder builder = OkHttpChannelBuilder.forAddress("127.0.0.1", serverPort)
    .overrideHostForAuthority("foo.test.google.fr");
```

**Comment by ejona86** [comment]
@madongfly LGTM

File: integration-testing/src/test/java/io/grpc/testing/integration/Http2OkHttpTest.java (hunk 3 above)
Comment (madongfly) [inline integration-testing/src/test/java/io/grpc/testing/integration/Http2OkHttpTest.java]: Done.

**Pull request merged**